### PR TITLE
Fix mmseqs/createtsv and add nf-test

### LINKED ISSUE #5014: Fix mmseqs/createtsv to work also with mmseqs/taxonomy
### Have you checked the docs?

- [X] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [X] [nf-core modules documentation](https://nf-co.re/docs/contributing/modules)

### Description of the bug

At the moment the submodule `mmseqs/createtsv` only seems to work with teh result from `mmseqs/cluster` due to  this optional flag `db_target = db_target ?: "${db_query}" // optional argument db_target as in many cases, it's the same as db_query`. This flag makes it a must that there is a third argument which is not the case for working with `mmseqs/taxonomy` results. to turn the database result of `mmseqs/taxonomy` to tsv it only needs two arguments the query_db and the taxonomy_result_db. In my tests it would not accept the optional argument which assigns the query_db twice if no target_db is used. 

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by Darcy220606**
##PR checklist

Closes #5014  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/mmseqs/createtsv/main.nf [hunk 1]
```diff
@@ -22,11 +22,11 @@ process MMSEQS_CREATETSV {
 
     script:
     def args = task.ext.args ?: ''
-    def args2 = task.ext.args ?: "*.dbtype"
-    def args3 = task.ext.args ?: "*.dbtype"
-    def args4 = task.ext.args ?: "*.dbtype"
+    def args2 = task.ext.args ?: "*.dbtype" // database generated by mmyseqs/cluster
```
Comment (Joon-Klaps) [inline modules/nf-core/mmseqs/createtsv/main.nf:25]:
```suggestion
    def args2 = task.ext.args ?: "*.dbtype" // database generated by mmyseqs cluster | search | taxonomy | ...
```

**Review by Joon-Klaps — APPROVED** [review]
Well spotted! Great work, small suggestions. You can merge if tests pass 🚀

File: modules/nf-core/mmseqs/createtsv/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    test("mmseqs/createtsv - bacteroides_fragilis - cluster") {
+
+        setup {
+            run("MMSEQS_CREATEDB", alias: "MMSEQS_QUERY") {
+                script "modules/nf-core/mmseqs/createdb/main.nf"
+                process {
+                    """
+                    input[0] = [
+                        [ id:'test_query', single_end:false ],
+                        file(params.test_data['bacteroides_fragilis']['genome']['genome_fna_gz'], checkIfExists: true),
```
Comment (Joon-Klaps) [inline modules/nf-core/mmseqs/createtsv/tests/main.nf.test:79]: Just noticed test failing due to memory issues. Consider using instead sars-cov-2 contigs for clustering.

File: modules/nf-core/mmseqs/createtsv/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    tag "mmseqs"
+    tag "mmseqs/taxonomy"
+    tag "mmseqs/cluster"
+    tag "mmseqs/createdb"
+    tag "mmseqs/databases"
+    tag "mmseqs/createtsv"
+
+    setup {
+        run("MMSEQS_CREATEDB", alias: "MMSEQS_TAXA") {
+            script "modules/nf-core/mmseqs/createdb/main.nf"
```
Comment (Joon-Klaps) [inline modules/nf-core/mmseqs/createtsv/tests/main.nf.test:19]: Sorry missed something, please make all the script locations relative.

**Review by Joon-Klaps — CHANGES_REQUESTED** [review]

**Comment by Darcy220606** [comment]
Hi @Joon-Klaps, so the cluster nf-tests would always run out of memory on github no matter what assertions i tried, so i used the tarr(ed) mmseqs testdataset files as you have done previously for this module pytests.  Do please have a look again at the PR and let me know if there is soemthing else to change. Thanks a bunch :)

**Review by Joon-Klaps — APPROVED** [review]
Very nice! LGTM! 🚀

**Pull request merged**